src/parser/parser_driver.cpp

```diff
@@ -1067,7 +1075,16 @@ void ParserDriver::defineGrammar()
 				std::shared_ptr<Symbol> parentSymbol = std::static_pointer_cast<const IdExpression>(iterable)->getSymbol();
 				assert(parentSymbol);
 				if (!parentSymbol->isArray())
-					error_handle((++args[3].getTokenIt())->getLocation(), "Identifier '" + parentSymbol->getName() + "' is not an array");
+				{
+					if (!incompleteMode() || !parentSymbol->isUndefined())
+						error_handle((++args[3].getTokenIt())->getLocation(), "Identifier '" + parentSymbol->getName() + "' is not an array");
+					else
+					{
+						auto parentExpr = std::static_pointer_cast<IdExpression>(iterable);
+						const auto& parentTokenText = parentExpr->getSymbolToken()->getText();
+						parentExpr->setSymbol(std::make_shared<ArraySymbol>(parentTokenText, ExpressionType::Undefined));
+					}
+				}
```

Comment:
Why so complicated? I needed the tormatted text to still look nice, which means I had to replace some `Symbol` base class instances with instances of particular derived classes. Also it was necessary to determine correct types of some Token instances (not everywhere).

Reply:
Good thing is, that the `IdExpression::setSymbol` method also updates the pointer stored in the `Token` associated with this `Symbol` instance.